src/router_config.c

```diff
@@ -26,6 +26,27 @@
 #include "entity_cache.h"
 #include "schema_enum.h"
 
+static void qdi_router_configure_body(qdr_core_t              *core,
+                                      qd_composed_field_t     *body,
+                                      qd_router_entity_type_t  type,
+                                      char                    *name)
+{
+    qd_buffer_list_t buffers;
+    qd_compose_take_buffers(body, &buffers);
+
+    qd_field_iterator_t *iter = qd_field_iterator_buffer(DEQ_HEAD(buffers), 0, qd_buffer_list_length(&buffers));
+    qd_parsed_field_t   *in_body = qd_parse(iter);
+    qd_field_iterator_free(iter);
+
+    qd_field_iterator_t *name_iter = 0;
+    if (name)
+        name_iter = qd_field_iterator_string(name);
+
+    qdr_manage_create(core, 0, type, name_iter, in_body, 0, buffers);
```

Comment:
What do you think about passing the buffers through the action so that the action method can free them when it is done?